lib/inspec/reporters/cli.rb

```diff
@@ -0,0 +1,396 @@
+# encoding: utf-8
+
+module Inspec::Reporters
+  class CLI < Base
+    case RUBY_PLATFORM
+    when /windows|mswin|msys|mingw|cygwin/
+      # Most currently available Windows terminals have poor support
+      # for ANSI extended colors
+      COLORS = {
+        'critical' => "\033[0;1;31m",
+        'major'    => "\033[0;1;31m",
+        'minor'    => "\033[0;36m",
+        'failed'   => "\033[0;1;31m",
+        'passed'   => "\033[0;1;32m",
+        'skipped'  => "\033[0;37m",
+        'reset'    => "\033[0m",
+      }.freeze
+
+      # Most currently available Windows terminals have poor support
+      # for UTF-8 characters so use these boring indicators
+      INDICATORS = {
+        'critical' => '[CRIT]',
+        'major'    => '[MAJR]',
+        'minor'    => '[MINR]',
+        'failed'   => '[FAIL]',
+        'skipped'  => '[SKIP]',
+        'passed'   => '[PASS]',
+        'unknown'  => '[UNKN]',
+      }.freeze
+    else
+      # Extended colors for everyone else
+      COLORS = {
+        'critical' => "\033[38;5;9m",
+        'major'    => "\033[38;5;208m",
+        'minor'    => "\033[0;36m",
+        'failed'   => "\033[38;5;9m",
+        'passed'   => "\033[38;5;41m",
+        'skipped'  => "\033[38;5;247m",
+        'reset'    => "\033[0m",
+      }.freeze
+
+      # Groovy UTF-8 characters for everyone else...
+      # ...even though they probably only work on Mac
+      INDICATORS = {
+        'critical' => '×',
+        'major'    => '∅',
+        'minor'    => '⊚',
+        'failed'   => '×',
+        'skipped'  => '↺',
+        'passed'   => '✔',
+        'unknown'  => '?',
+      }.freeze
+    end
+
+    MULTI_TEST_CONTROL_SUMMARY_MAX_LEN = 60
+
+    def render
+      run_data[:profiles].each do |profile|
+        @control_count = 0
+        output ''
+        print_profile_header(profile)
+        print_standard_control_results(profile)
+        print_anonymous_control_results(profile)
+        output format_message(
+          indentation: 5,
+          message: 'No tests executed.',
+        ) if @control_count == 0
+      end
+
+      output ''
+      print_profile_summary
+      print_tests_summary
+      @output
```

Comment:
This feels like a code smell that subclasses need to know to return an instance variable containing they output they've been creating (by calling a method that appends to an instance variable), which is pretty implementation-specific.

I'd encourage you to refactor this a bit. I'm thinking something needs to change in `Inspec::Reporters.render` so it looks something like this:

```ruby
def self.render(config)
  reporter = case config[:name]
             when 'cli'
               Inspec::Reporters::CLI.new(config).render
             when 'json'
               Inspec::Reporters::Json.new(config).render
             # etc...
             end
  reporter.render # does the actual rendering of text and storage, and should not attempt to do it again if we've already stored rendered text
  reporter.rendered_output # returns the actual output, which can be called multiple times if eneded
end
```